src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -0,0 +1,84 @@
+package kitchenpos.menu.application;
+
+import static kitchenpos.exception.KitchenposExceptionMessage.NOT_FOUND_MENU_GROUP;
+import static kitchenpos.exception.KitchenposExceptionMessage.NOT_FOUND_PRODUCT;
+import static kitchenpos.exception.KitchenposExceptionMessage.PRICE_CANNOT_LOWER_THAN_MIN;
+
+import java.util.stream.Collectors;
+import kitchenpos.exception.KitchenposException;
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.dto.MenuProductRequest;
+import kitchenpos.menu.dto.MenuRequest;
+import kitchenpos.menu.dto.MenuResponse;
+import kitchenpos.menugroup.domain.MenuGroup;
+import kitchenpos.menugroup.domain.MenuGroupRepository;
+import kitchenpos.product.domain.ProductRepository;
+import kitchenpos.menu.domain.MenuProduct;
+import kitchenpos.product.domain.Product;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+@Transactional
+@Service
+public class MenuService {
+
+    private static final BigDecimal MIN_PRICE = BigDecimal.ZERO;
+
+    private final MenuRepository menuRepository;
+    private final MenuGroupRepository menuGroupRepository;
+    private final ProductRepository productRepository;
+
+    public MenuService(final MenuRepository menuRepository,
+                       final MenuGroupRepository menuGroupRepository,
+                       final ProductRepository productRepository) {
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.productRepository = productRepository;
+    }
+
+    public MenuResponse create(final MenuRequest request) {
+        checkPriceGreaterThanMin(request.getPrice());
+        final MenuGroup menuGroup = findMenuGroupById(request.getMenuGroupId());
+        return MenuResponse.of(menuRepository.save(request.toMenu(menuGroup,
+                                                                  getMenuProducts(request))));
+    }
+
+    private List<MenuProduct> getMenuProducts(MenuRequest request) {
+        return request.getMenuProducts()
+                      .stream()
+                      .map(this::createMenuProduct)
+                      .collect(Collectors.toList());
+    }
+
+    private MenuProduct createMenuProduct(MenuProductRequest menuProductRequest) {
+        return new MenuProduct(findProductById(menuProductRequest),
+                               menuProductRequest.getQuantity());
+    }
+
+    private Product findProductById(MenuProductRequest menuProductRequest) {
+        return productRepository.findById(menuProductRequest.getProductId())
+                                .orElseThrow(() -> new KitchenposException(NOT_FOUND_PRODUCT));
+    }
```

Comment:
조회만 하는 경우 @Transactional(readOnly = true)를 통해 소소한 성능 개선을 할 수 있어요! 😄 
```suggestion
    @Transactional(readOnly = true)
    private Product findProductById(MenuProductRequest menuProductRequest) {
        return productRepository.findById(menuProductRequest.getProductId())
                                .orElseThrow(() -> new KitchenposException(NOT_FOUND_PRODUCT));
    }
```

Reply:
지금 상태로는 해당 메서드가 create 메서드 내부에서 호출되는 메서드라 별도의 트랜잭션 적용은 힘들 것 같습니다
readOnly를 적용하려면 product 조회 기능을 다른 서비스를 주입받아 호출하거나 Repo를 감싼 DAO를 생성해야할 것 같은데 혹시 추가적으로 주실 의견 있으실까요~??

ps
ProductService쪽으로 findProductById를 옮기고 MenuService에서 ProductService를 주입받는 방식도 생각해봤는데 같은 계층 서비스가 다른 서비스를 의존하는 것이 좋은 건지 의문이 듭니다..

Reply:
앗 이부분은 제가 잘못 봤네요! 🙏 
MenuService 에서 ProductService 를 주입받는 것은 지양하는 것이 좋다고 생각합니다!
지금 그대로 진행해시면 될 것 같아요!
혼선을 드려 죄송합니다 🙇‍♀️